pkg/controller/volume/persistentvolume/scheduler_binder_test.go

```diff
@@ -1296,20 +1324,21 @@ func TestBindPodVolumes(t *testing.T) {
 		},
 		"nothing-to-bind-empty": {},
 		"already-bound": {
-			binding:   binding1aBound,
-			cachedPV:  pvNode1aBound,
-			cachedPVC: boundPVCNode1a,
+			binding:    binding1aBound,
+			cachedPVs:  []*v1.PersistentVolume{pvNode1aBound},
+			cachedPVCs: []*v1.PersistentVolumeClaim{addProvisionAnn(boundPVCNode1a)},
 		},
 		"binding-succeeds-after-time": {
-			binding:   binding1aBound,
-			cachedPV:  pvNode1a,
-			cachedPVC: unboundPVC,
+			binding:    binding1aBound,
+			cachedPVs:  []*v1.PersistentVolume{pvNode1a},
+			cachedPVCs: []*v1.PersistentVolumeClaim{unboundPVC},
 			delayFunc: func(t *testing.T, testEnv *testEnv, pod *v1.Pod, pv *v1.PersistentVolume, pvc *v1.PersistentVolumeClaim) {
 				// Update PVC to be fully bound to PV
 				newPVC := pvc.DeepCopy()
 				newPVC.ResourceVersion = "100"
 				newPVC.Spec.VolumeName = pv.Name
 				metav1.SetMetaDataAnnotation(&newPVC.ObjectMeta, annBindCompleted, "yes")
+				metav1.SetMetaDataAnnotation(&newPVC.ObjectMeta, annSelectedNode, pod.Spec.NodeName)
```

Comment:
is this necessary?

Reply:
Yes, otherwise it'd fail in [this condition](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/persistentvolume/scheduler_binder.go#L419).

It wasn't necessary before because the execution never made it inside [that loop](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/persistentvolume/scheduler_binder.go#L408-L426) (the pod binding cache wasn't updated, resulting in an emtpy `claimsToProvision`).